bn-apps/memberships-management/membership-service/src/main/kotlin/net/corda/businessnetworks/membership/bno/ActivateMembershipFlow.kt

```diff
@@ -48,4 +47,40 @@ class ActivateMembershipFlow(val membership : StateAndRef<Membership.State>) : F
 
         return stx
     }
+}
+
+/**
+ * This is a convenience flow that can be easily used from command line
+ *
+ * @param party whose membership state to be activated
+ */
+@InitiatingFlow
+@StartableByRPC
+class ActivateMembershipForPartyFlow(val party : Party) : FlowLogic<SignedTransaction>() {
+
+    companion object {
+        object LOOKING_FOR_MEMBERSHIP_STATE : ProgressTracker.Step("Looking for party's membership state")
+        object ACTIVATING_THE_MEMBERSHIP_STATE : ProgressTracker.Step("Activating the membership state")
+
+        fun tracker() = ProgressTracker(
+                LOOKING_FOR_MEMBERSHIP_STATE,
+                ACTIVATING_THE_MEMBERSHIP_STATE
+        )
+    }
+
+    override val progressTracker = tracker()
+
+    @Suspendable
+    override fun call() : SignedTransaction {
+        progressTracker.currentStep = LOOKING_FOR_MEMBERSHIP_STATE
+        val stateToActivate = findMembershipStateForParty(party)
+
+        progressTracker.currentStep = ACTIVATING_THE_MEMBERSHIP_STATE
+        return subFlow(ActivateMembershipFlow(stateToActivate))
+    }
+
+    private fun findMembershipStateForParty(party : Party) : StateAndRef<Membership.State> {
+        //@todo this could be made more effective and look for the Party's state in the vault
+        return serviceHub.vaultService.queryBy<Membership.State>().states.filter { it.state.data.member == party }.single()
```

Comment:
As this flow is to be executed from a command line, if the state has not been found in the vault, lets throw some more meningful exception i.e. like `FlowException("State for party $party has not been found")`.

Reply:
ok